### DISCUSSION TITLE
Order Demand for delivery locations

### DISCUSSION THREAD

**Discussion by AltaherPresto** [google/or-tools#3174] [Routing (and legacy CP) questions] [answered]
I'm using the OR-Tools VRP solver to solve a [Capacitated Vehicle Routing Problem](https://developers.google.com/optimization/routing/cvrp) with [Pickup-Delivery constraints](https://developers.google.com/optimization/routing/pickup_delivery)

- Pickup index must be visited before its corresponding delivery index
- Each pickup/delivery index pair must be assigned to the same vehicle

I set the order demand value to `1` for all pickup locations and `0` for all delivery locations.
**Is it correct to set demand for delivery locations to 0?**
Shouldn't it be `-1` since the vehicle (at the delivery location) is offloading the demand from the pickup location and thus vacating part of their vehicle capacity that can (potentially) be used for other pickups along its route?

I ask because the docs don't cover CVRP in the context of pickup/delivery

**Comment by jmarca** [comment, marked as answer]
In this case, demand is -1.

Dimensions are just accumulators.  You add, and sometimes subtract, depending on the problem.  The only hiccup is that they are strictly positive, so if you expect you can dip below zero you will have errors.

Once I thought to model breaks by dipping my "need a break" accumulator below zero and that failed badly.
